backend/api/routes/connections.js

```diff
@@ -150,4 +150,39 @@ router
     }
   });
 
+router.route("/send-request").post(authenticateMiddleware, async (req, res) => {
+  try {
+    const { receiver_id } = req.body;
+    const sender_id = req.user.id;
+    const supabaseClient = req.supabase;
+
+    //check if the request being sent out exists in our table already jic
+    const { data: existingRequest } = await supabaseClient
+      .from("connection_requests")
+      .select("*")
+      .eq("sender_id", sender_id)
+      .eq("receiver_id", receiver_id)
+      .single();
+
+    if (existingRequest) {
+      return res.status(400).json({ error: "Request already sent" });
+    }
+
+    //since we already checked if the connection request existed previously add the actual connection request now with status pending. We will change it to approved when user clicks accept on their end
+    const { error } = await supabaseClient.from("connection_requests").insert({
+      sender_id: sender_id,
+      receiver_id: receiver_id,
+      status: "pending",
```

Comment:
Ditto before, you're thinking of this tables as non-relational ones. This status field should be on the connections table, you can initialize all of them with "unsent", then you can do a flow like:
```
unsent -> pending               -> accepted -> deleted
       -> not interested
                                -> rejected
```
This way you can manage everything in one place and avoid doing so many queries